### PR TITLE
Move dependency related classes to Transactron

### PR DESCRIPTION
**Pull request by tilk**
Another change to make #538 cleaner. Assertions are a general concept, the implementation should go to library code.

### CODE REVIEW

**Review by lekcyjna123 — APPROVED** [review]

**Comment by tilk** [comment]
Did a compromise and moved `UnifierKey` to `lib`, but the rest stays in `utils`. Maybe not perfect, but splits the transaction-dependent code from the transaction-independent.

File: transactron/lib/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from ..utils.dependencies import *
+
+
+__all__ = [
+    "DependencyManager",
+    "DependencyKey",
+    "SimpleKey",
+    "ListKey",
+    "UnifierKey"
+]
```
[the remark this answers is not in the document]
Comment (tilk) [reply transactron/lib/dependencies.py:14]: Unfortunately, pyright generates a warning for such an use of `__all__`. I'm not happy about this, too.

[the remark this answers is not in the document]
Comment (piotro888) [reply transactron/lib/dependencies.py:14]: I found that Amaranth does that trick with `__all__` in enums https://github.com/amaranth-lang/amaranth/blob/ded84fe9d640c1fffa4a089fcb7233319da5ea93/amaranth/lib/enum.py#L9 but it doesn't look too pretty. I leave the decision up to you.

**Review by piotro888 — APPROVED** [review]

File: transactron/lib/dependencies.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tilk) [reply transactron/lib/dependencies.py:14]: Our stub file for `amaranth.lib.enum` doesn't use that trick because pyright doesn't like it.

**Comment by piotro888** [comment]
Something interesting happened on Basic core device utilization...
image
(Fmax also dropped because of this)

**Comment by tilk** [comment]
What. How could this happen? There are no functional changes here at all.

**Comment by lekcyjna123** [comment]
And what is interesting, the change on Full configuration is much smaller (+2000 LUT vs +15 000 LUT in Basic).

**Pull request merged**